### PR TITLE
E2E test assertions

### LINKED ISSUE #4: Add assertions to end-to-end test script
Currently, we have a simple `e2e-test.js` script which creates two deposits and redeems funds from the first one. However, we don't check any expectations regarding account balances. It would be nice to add them and fail in case something goes wrong.

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Closes #4 
Depends on #11 

Here we introduce some assertions checking TBTC and BTC balances during the execution of the end-to-end tests.

### CODE REVIEW

File: e2e/assertions.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    const actualTbtcBalanceBn = await getTbtcAccountBalance(web3, tbtc, account)
+    const expectedTbtcBalanceBn = web3.utils.toBN(expectedTbtcBalance)
+
+    if (!actualTbtcBalanceBn.eq(expectedTbtcBalanceBn)) {
+        throw new Error(
+            `unexpected TBTC balance on account ${account}:
+                actual:   ${actualTbtcBalanceBn}
+                expected: ${expectedTbtcBalanceBn}`
+        )
+    }
```
Comment (nkuba) [inline e2e/assertions.js:31]: I was thinking about using `chai` for assertions to simplify comparisons and error messages.

File: e2e/assertions.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    const expectedTbtcAmountBn = web3.utils.toBN(expectedTbtcAmount)
+
+    if (!actualTbtcAmountBn.eq(expectedTbtcAmountBn)) {
+        throw new Error(
+            `unexpected minted TBTC amount:
+                actual:   ${actualTbtcAmountBn}
+                expected: ${expectedTbtcAmountBn}`
+        )
+    }
+}
```
Comment (nkuba) [inline e2e/assertions.js:14]: Maybe we don't need to extract it to a separate function and do this check directly in test code. I was a little bit lost with the wrapper for the deposit we've defined for deposit object from tbtc.js.

File: e2e/e2e-test.js [hunk 3]
```diff
@@ -5,10 +5,18 @@ import ProviderEngine from "web3-provider-engine"
...
+    assertTbtcAccountBalance
+} from "./assertions.js";
+import {getTbtcAccountBalance} from "./common.js";
 
 const depositsCount = 2
 const satoshiLotSize = 100000 // 0.001 BTC
 const btcAddress = '2N6L4Q6fphMzuWqERQYTwgEMmQTpcqgdVFK'
+const signerFeeDivisor = 0.0005 // 0.05%
+const tbtcDepositAmount = 1000000000000000 // satoshiLotSize * satoshiMultiplier
+const tbtcDepositAmountMinusSignerFee = (1 - signerFeeDivisor) * tbtcDepositAmount
```
Comment (nkuba) [inline e2e/e2e-test.js:19]: We should also confirm the signer fee went to an expected address.

File: e2e/e2e-test.js [hunk 4]
```diff
@@ -51,18 +59,67 @@ async function run() {
...
+        `performing redemption should be: ${afterRedemptionTbtcAccountBalance}. ` +
+        `Checking assertion...`
+    )
+
+    await assertTbtcAccountBalance(
+        web3,
+        tbtc,
+        web3.eth.defaultAccount,
+        afterRedemptionTbtcAccountBalance
+    )
```
Comment (nkuba) [inline e2e/e2e-test.js:122]: I'd add bitcoin address balance check after redemption.

File: e2e/assertions.js (hunk 1 above)
Comment (lukasz-zimnoch) [reply e2e/assertions.js:31]: I tried to do this but then we have to run those tests using a testing framework.

File: e2e/assertions.js (hunk 2 above)
Comment (lukasz-zimnoch) [reply e2e/assertions.js:14]: I think we will need to do the same in further tests hence I extracted this assertion.

File: e2e/e2e-test.js (hunk 4 above)
Comment (nkuba) [reply e2e/e2e-test.js:122]: Do you think it's achievable? Can you cover it in this PR or do we want to handle it in a separate PR?

Comment (lukasz-zimnoch) [reply e2e/e2e-test.js:121]: May require some juggling with BTC wallets but I'll try.

File: e2e/e2e-test.js (hunk 3 above)
Comment (lukasz-zimnoch) [reply e2e/e2e-test.js:21]: Ok, I'll try to add this.

File: e2e/common.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+export const getReceivedBtcAmount = async (bitcoinRpc, address) => {
+    const received = (await bitcoinRpc.listreceivedbyaddressAsync(1, false, true, address)).result
+
+    let receivedBtcAmount = 0
+
+    for (let i = 0; i < received.length; i++) {
+        if (received[i].address === address) {
+            receivedBtcAmount += received[i].amount
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply e2e/common.js:16]: Nope, they are often floating points numbers which can't be handled by BN.

**Pull request merged**